packages/commonwealth/server/controllers/server_admin_methods/get_top_users.ts

```diff
@@ -0,0 +1,90 @@
+import { AppError } from '@hicommonwealth/adapters';
+import { UserInstance } from '@hicommonwealth/model';
+import { QueryTypes } from 'sequelize';
+import { ServerAdminController } from '../server_admin_controller';
+
+export const Errors = {
+  NotAdmin: 'Must be a site admin',
+};
+
+export type GetTopUsersOptions = {
+  user: UserInstance;
+};
+
+export type GetTopUsersResult = {
+  top_activity: number;
+  thread_count: number;
+  comment_count: number;
+  user_id: number;
+  profile_id: number;
+  profile_name: string;
+  address_ids: number[];
+  community_ids: string[];
+}[];
+
+export async function __getTopUsers(
+  this: ServerAdminController,
+  { user }: GetTopUsersOptions,
+): Promise<GetTopUsersResult> {
+  if (!user.isAdmin) {
+    throw new AppError(Errors.NotAdmin);
+  }
+
+  const sql = `
+    WITH UserActivity AS (
+        SELECT
+            u.id AS user_id,
+            p.id AS profile_id,
+            a.id AS address_id,
+            p.profile_name as profile_name,
+            a.community_id,
+            COUNT(DISTINCT t.id) AS thread_count,
+            COUNT(DISTINCT c.id) AS comment_count
+        FROM
+            "Users" u
+        LEFT JOIN "Profiles" p ON u.id = p.user_id
+        LEFT JOIN "Addresses" a ON u.id = a.user_id
+        LEFT JOIN "Threads" t ON a.id = t.address_id
+        LEFT JOIN "Comments" c ON a.id = c.address_id
+        GROUP BY
+            u.id, p.id, p.profile_name, a.id, a.community_id
+    ),
+    RankedUsers AS (
+        SELECT
+            user_id,
+            profile_id,
+            profile_name,
+            thread_count,
+            comment_count,
+            array_agg(address_id) AS addresses_ids,
+            array_agg(community_id) AS community_ids,
+            SUM(thread_count + comment_count) AS total_activity
+        FROM
+            UserActivity
+        GROUP BY
+            user_id, profile_id, profile_name, thread_count, comment_count
+      HAVING
+          NOT ('edgeware' = ANY(array_agg(community_id)) OR 'cmn-protocol' = ANY(array_agg(community_id)))
+      ORDER BY
+        total_activity DESC
+    )
+    SELECT
+        total_activity,
+        thread_count,
+        comment_count,
+        user_id,
+        profile_id,
+        profile_name,
+        array_to_string(addresses_ids, '|') as address_ids,
+        array_to_string(community_ids, '|') as community_ids
```

Comment:
community_ids looks to be broken, it only returns a single community_id for each user. I will look into this further and respond to this comment if I find the breakage.

Reply:
I think it's just that the top users tend to only use a single community.

I increased the limit of the query to show more results and saw a few that had multiple addresses and multiple communities.

Reply:
Does that query consider the filtering out of users related to edgeware/cmn-protocol?

Reply:
I mean run this top_users query. Find the top user (redacted for privacy). Then plug the user_id into the query in my comment above. Then you get multiple communities as well as multiple addresses of the associated community

Reply:
Got it, thanks for digging into it!

Reply:
So I can't actually figure out what is wrong with this particular query, but it seems this similar query pulls the correct community/address ids:

```
SELECT 
  p.profile_name AS profile_name,
  COUNT(DISTINCT t.id) AS thread_count,
  COUNT(DISTINCT c.id) AS comment_count,
  COUNT(DISTINCT c.id) + COUNT(DISTINCT t.id) AS total_count,
  ARRAY_AGG(DISTINCT a.community_id ORDER BY a.community_id) AS community_ids,
  ARRAY_AGG(DISTINCT a.id ORDER BY a.id) AS address_ids
FROM "Profiles" AS p
JOIN "Addresses" AS a ON p.id = a.profile_id
LEFT JOIN "Threads" AS t ON a.id = t.address_id
LEFT JOIN "Comments" AS c ON a.id = c.address_id
GROUP BY p.id
ORDER BY TOTAL_COUNT DESC
LIMIT 150;
```

You will still need to add the extra conditions (filter out commonwealth employees), but this might be a better base query to work off for now